src/assets/wise5/services/notificationService.ts

```diff
@@ -474,4 +474,34 @@ export class NotificationService {
   broadcastViewCurrentAmbientNotification(args: any) {
     this.viewCurrentAmbientNotificationSource.next(args);
   }
+
+  showSavingMessage(): void {
+    this.broadcastSetGlobalMessage({
+      globalMessage: {
+        text: $localize`Saving...`,
+        isProgressIndicatorVisible: true,
+        time: null
+      }
+    });
+  }
+
+  /*
+   * @param message The message to show.
+   * @param timeout The amount of time to wait in milliseconds before changing the message from
+   * 'Saving...' to 'Saved' assuming showSavingMessage() was previously called. We do this to make
+   * sure the 'Saving...' message is visible long enough for the user to notice otherwise it might
+   * switch too fast from 'Saving...' to 'Saved'. If we don't perform the wait, it will always say
+   * 'Saved' and the author might wonder whether anything was ever saved.
+   */
```

Comment:
I think we can remove the rationale in the explanation for the `timeout` param here, since it references a specific use case not related to the component itself. I'm actually wondering if we need the @param documentations for this function at all, as the parameter names seem self-explanatory to me.